js/source/vector_tile_source.js

```diff
@@ -71,9 +72,15 @@ VectorTileSource.prototype = util.inherit(Evented, {
             overscaling: overscaling,
             angle: this.map.transform.angle,
             pitch: this.map.transform.pitch,
-            collisionDebug: this.map.collisionDebug
+            collisionDebug: this.map.collisionDebug,
+            parentId: tile.parentId
         };
 
+        // request the tile parentID if it exists
+        if (tile.parentId) {
```

Comment:
Right. If a tile has a parentId the url for the tile request needs to be used in place of the url built from `tile.id`